### PR TITLE
[CELEBORN-711][TEST] Rework PushDataTimeoutTest

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

1. separated push data timeout tests and push merge data timeout tests in `PushDataTimeoutTest`
2. updated the test results assertion
3. rework `pushdata timeout will add to blacklist`

### Why are the changes needed?

ensure that the timeout behavior is correctly implemented

https://github.com/apache/incubator-celeborn/pull/1613#discussion_r1236423721

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

existing tests

### CODE REVIEW

File: tests/spark-it/src/test/scala/org/apache/celeborn/tests/spark/PushDataTimeoutTest.scala [hunk 1]
```diff
@@ -66,27 +100,23 @@ class PushDataTimeoutTest extends AnyFunSuite
...
+      }
     }
   }
 
   test("celeborn spark integration test - pushdata timeout will add to balcklist") {
     val sparkConf = new SparkConf().setAppName("rss-demo").setMaster("local[2]")
       .set(s"spark.${CelebornConf.CLIENT_PUSH_DATA_TIMEOUT.key}", "5s")
-      .set(s"spark.celeborn.data.push.timeoutCheck.interval", "2s")
-      .set(s"spark.${CelebornConf.CLIENT_BLACKLIST_SLAVE_ENABLED.key}", "false")
+      .set(s"spark.${CelebornConf.CLIENT_BLACKLIST_SLAVE_ENABLED.key}", "true")
```
Comment (cfmcgrady) [inline tests/spark-it/src/test/scala/org/apache/celeborn/tests/spark/PushDataTimeoutTest.scala:119]: this should be accidentally changed by https://github.com/apache/incubator-celeborn/pull/1501

File: tests/spark-it/src/test/scala/org/apache/celeborn/tests/spark/PushDataTimeoutTest.scala [hunk 2]
```diff
@@ -34,30 +36,62 @@ class PushDataTimeoutTest extends AnyFunSuite
     logInfo("test initialized , setup celeborn mini cluster")
     val workerConf = Map(
       CelebornConf.TEST_CLIENT_PUSH_MASTER_DATA_TIMEOUT.key -> "true",
-      CelebornConf.TEST_WORKER_PUSH_SLAVE_DATA_TIMEOUT.key -> "true",
-      "celeborn.push.timeoutCheck.interval" -> "1s")
```
Comment (cfmcgrady) [inline tests/spark-it/src/test/scala/org/apache/celeborn/tests/spark/PushDataTimeoutTest.scala:38]: no documentation or usage information could be found for the `celeborn.push.timeoutCheck.interval` parameter.

File: tests/spark-it/src/test/scala/org/apache/celeborn/tests/spark/PushDataTimeoutTest.scala [hunk 3]
```diff
@@ -34,30 +36,62 @@ class PushDataTimeoutTest extends AnyFunSuite
...
     System.gc()
   }
 
-  test("celeborn spark integration test - pushdata timeout") {
-    Seq("false", "true").foreach { enabled =>
+  Seq(false, true).foreach { enabled =>
+    test(s"celeborn spark integration test - pushdata timeout w/ replicate = $enabled") {
       val sparkConf = new SparkConf().setAppName("rss-demo").setMaster("local[2]")
         .set(s"spark.${CelebornConf.CLIENT_PUSH_DATA_TIMEOUT.key}", "5s")
-        .set(s"spark.celeborn.data.push.timeoutCheck.interval", "2s")
```
Comment (cfmcgrady) [inline tests/spark-it/src/test/scala/org/apache/celeborn/tests/spark/PushDataTimeoutTest.scala:54]: the same with `celeborn.push.timeoutCheck.interval`

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 4]
```diff
@@ -133,13 +131,13 @@ class PushDataHandler extends BaseMessageHandler with Logging {
...
-      pushMasterDataTimeoutTested = true
+    if (isMaster && testPushMasterDataTimeout && !PushDataHandler.pushMasterDataTimeoutTested.getAndSet(
+        true)) {
       return
     }
 
-    if (!isMaster && testPushSlaveDataTimeout && !pushSlaveDataTimeoutTested) {
-      pushSlaveDataTimeoutTested = true
+    if (!isMaster && testPushSlaveDataTimeout && !PushDataHandler.pushSlaveDataTimeoutTested.getAndSet(
+        true)) {
```
Comment (pan3793) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:140]:
```suggestion
    if (!isMaster && testPushSlaveDataTimeout && 
        !PushDataHandler.pushSlaveDataTimeoutTested.getAndSet(true)) {
```

File: tests/spark-it/src/test/scala/org/apache/celeborn/tests/spark/PushDataTimeoutTest.scala [hunk 5]
```diff
@@ -98,6 +136,21 @@ class PushDataTimeoutTest extends AnyFunSuite
...
+    val blacklist = SparkContextHelper.env
+      .shuffleManager
+      .asInstanceOf[RssShuffleManager]
+      .getLifecycleManager
+      .workerStatusTracker
+      .blacklist
+
+    assert(blacklist.size() > 0)
+    blacklist.asScala.foreach {
+      case (_, (code, _)) =>
```
Comment (pan3793) [inline tests/spark-it/src/test/scala/org/apache/celeborn/tests/spark/PushDataTimeoutTest.scala:150]:
```suggestion
    blacklist.asScala.foreach { case (_, (code, _)) =>
```

**Review by pan3793 — APPROVED** [review]

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala (hunk 4 above)
Comment (cfmcgrady) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala]: `spotless` reformat this to 2 spaces instead 4.

**Comment by pan3793** [comment]
Merged to main/0.3

**Pull request closed without merging**